### PR TITLE
Adding passport local authentication in index.js 

### PR DESCRIPTION
**Pull request by Shashankpantiitbhilai**
Migrated to passport authentication, implemented sessions and "login" and "register" route 
@krishnan05  @DevilsAutumn

### CODE REVIEW

**Comment by DevilsAutumn** [comment]
Can we make the code little modular ? For example: by moving the schema from index.js to the folder backend/models, the logic of initializing passport to backend/passport.js, isAuthenticated to backend/middlewares and routes to backend/routes .

**Comment by Shashankpantiitbhilai** [comment]
ok i will look into that

**Comment by Shashankpantiitbhilai** [comment]
I am getting this error.
Screenshot 2024-01-21 202840

I have already searched in the stackoverflow and other modes but did not resolve
there is some error with the parsing error

**Comment by DevilsAutumn** [comment]
@Shashankpantiitbhilai  please make sure you are not using const as a variable name.

**Comment by Shashankpantiitbhilai** [comment]
No I used const for importing the modules

**Comment by DevilsAutumn** [comment]
@Shashankpantiitbhilai please check out this link: https://stackoverflow.com/questions/42706584/eslint-error-parsing-error-the-keyword-const-is-reserved
Looks like the eslint is not configured properly.

**Comment by Shashankpantiitbhilai** [comment]
I have pushed the changes now

File: Backend/index.js [hunk 1]
```diff
@@ -1,40 +1,37 @@
+/* eslint-env es6 */
+/* eslint-disable */
```
Comment (DevilsAutumn) [inline Backend/index.js:2]: why this?

File: Backend/routes/auth.js [hunk 2]
```diff
@@ -5,39 +5,36 @@
...
+        });
+    } catch (error) {
+      console.error(error);
+      return res.status(500).json({ success: false, message: "Internal Server Error" });
     }
-  );
-
+  });
 
-  router.post('/add', restrictToPresident, async (req, res) => {
```
Comment (DevilsAutumn) [inline Backend/routes/auth.js:40]: why `restrictToPresident` is removed?

File: Backend/routes/auth.js [hunk 3]
```diff
@@ -152,7 +149,7 @@
   });
 
 
-  router.post('/update', restrictToAdmin, async (req, res) => {
```
Comment (DevilsAutumn) [inline Backend/routes/auth.js:155]: why `restrictToAdmin` is removed?

**Comment by DevilsAutumn** [comment]
Looks like as of now it supports just normal id pass login and not using google oauth. Do we have a separate PR coming for that?

**Comment by Shashankpantiitbhilai** [comment]
I have added google oauth now and also fixed the previous errors

**Comment by Snehil-Shah** [comment]
Current State:
Login & Registration works with Local Strategy using the UI

To-Do:
- Integrate Google Auth Strategy with the frontend (Snehil)
- Implement User Sessions (Snehil)
- OTP Email Confirmation (Shashank)

**Comment by Snehil-Shah** [comment]
Current Progress: @DevilsAutumn & @krishnan05  (*Give some time for the GIFs to load*)

Register Page:
register_screen

Logging in and user session:
persistent_login

Google Authentication:
google

**Comment by DevilsAutumn** [comment]
Looks great!

**Comment by Snehil-Shah** [comment]
@DevilsAutumn @krishnan05 this PR is ready for final review

**Comment by krishnan05** [comment]
> @DevilsAutumn @krishnan05 this PR is ready for final review

I am unable to test the google login, as "Student Database COSA has not completed the Google verification process. The app is currently being tested, and can only be accessed by developer-approved testers." So please add my gmail ayush05012003krishn@gmail.com account so that i can test it. I have tested the passport login and it works well. One suggestion from my end would be to also add logout feature in this PR only. Moreover as i logged in as a student, i shouldn't be getting add user feature

**Comment by Snehil-Shah** [comment]
> > @DevilsAutumn @krishnan05 this PR is ready for final review
> 
> I am unable to test the google login, as "Student Database COSA has not completed the Google verification process. The app is currently being tested, and can only be accessed by developer-approved testers." So please add my gmail [ayush05012003krishn@gmail.com](mailto:ayush05012003krishn@gmail.com) account so that i can test it. I have tested the passport login and it works well. One suggestion from my end would be to also add logout feature in this PR only. Moreover as i logged in as a student, i shouldn't be getting add user feature

Added your email.
Logout feature was planned on week 3, so added a basic Logout link for testing right now.

About Add User feature, we will be tackling that in week 3 when working on permission handling.

**Comment by krishnan05** [comment]
> > > @DevilsAutumn @krishnan05 this PR is ready for final review
> > 
> > 
> > I am unable to test the google login, as "Student Database COSA has not completed the Google verification process. The app is currently being tested, and can only be accessed by developer-approved testers." So please add my gmail [ayush05012003krishn@gmail.com](mailto:ayush05012003krishn@gmail.com) account so that i can test it. I have tested the passport login and it works well. One suggestion from my end would be to also add logout feature in this PR only. Moreover as i logged in as a student, i shouldn't be getting add user feature
> 
> @krishnan05 Added your email. Logout feature was planned on week 3, so added a basic Logout link for testing right now. (Sync your local branch)
> 
> About Add User feature, we will be tackling that in week 3 when working on permission handling.

Okay. 
I have tested g-auth and its working
Good

**Review by krishnan05 — APPROVED** [review]

**Comment by Snehil-Shah** [comment]
If reviewed, can you merge this branch? we don't want to create many branches to avoid unnecessary conflicts so will be reusing this branch.

**Pull request merged**